tests/unit/lms/views/basic_lti_launch_test.py

```diff
@@ -241,54 +240,27 @@ def test_it_adds_the_document_url(self, context, pyramid_request):
 
 
 class TestUnconfiguredBasicLTILaunch:
-    def test_it_sets_the_right_javascript_config_settings(
+    def test_it_enables_content_item_selection_mode(
         self, BearerTokenSchema, bearer_token_schema, context, pyramid_request
     ):
-        pyramid_request.params.update({"some_random_rubbish": "SOME_RANDOM_RUBBISH"})
-
-        pyramid_request.registry.settings["google_client_id"] = "TEST_GOOGLE_CLIENT_ID"
-        pyramid_request.registry.settings[
-            "google_developer_key"
-        ] = "TEST_GOOGLE_DEVELOPER_KEY"
-
         BasicLTILaunchViews(context, pyramid_request).unconfigured_basic_lti_launch()
 
-        assert context.js_config.config == {
-            "enableLmsFilePicker": False,
-            "formAction": "http://example.com/module_item_configurations",
-            "formFields": Any.dict(),
-            "googleClientId": "TEST_GOOGLE_CLIENT_ID",
-            "googleDeveloperKey": "TEST_GOOGLE_DEVELOPER_KEY",
-            "customCanvasApiDomain": context.custom_canvas_api_domain,
-            "lmsUrl": context.lms_url,
-            "urls": {},
-        }
-
-        form_fields = context.js_config.config["formFields"]
-
-        # Test that we pass through parameters from the request made to us
-        # onto the configure module item call
         BearerTokenSchema.assert_called_once_with(pyramid_request)
         bearer_token_schema.authorization_param.assert_called_once_with(
             pyramid_request.lti_user
         )
-        assert form_fields == Any.dict.containing(
-            {
-                "authorization": bearer_token_schema.authorization_param.return_value,
+        context.js_config.enable_content_item_selection_mode.assert_called_once_with(
+            form_action="http://example.com/module_item_configurations",
+            form_fields={
+                "user_id": "TEST_USER_ID",
                 "resource_link_id": "TEST_RESOURCE_LINK_ID",
-                "tool_consumer_instance_guid": "TEST_TOOL_CONSUMER_INSTANCE_GUID",
                 "oauth_consumer_key": "TEST_OAUTH_CONSUMER_KEY",
-                "user_id": "TEST_USER_ID",
+                "tool_consumer_instance_guid": "TEST_TOOL_CONSUMER_INSTANCE_GUID",
                 "context_id": "TEST_CONTEXT_ID",
-                "some_random_rubbish": "SOME_RANDOM_RUBBISH",
-            }
+                "authorization": bearer_token_schema.authorization_param.return_value,
```

Comment:
Not easy to see in the diff, but there's a lot of duplication here with the form fields from the fixture. I think this might benefit from a `form_fields` fixture used by both.

There's also a testing gap here. We don't check that we remove any of the `oauth_nonce` and friends.

Reply:
Ah yes, because the `pyramid_request` fixture doesn't have the `oauth` params in the first place. I'll add them.

I think the duplication can be removed by doing `"user_id": pyramid_request.params["user_id"]` etc instead of `"user_id": "TEST_USER_ID"`. I'm not sure that adds much value with "sentinel strings" like `"TEST_USER_ID"` but I'll do it anyway. It sort of makes it clearer what it's testing

Reply:
This is better, but it's not quite what I meant. Not worth going over again I'd say, but I was thinking something like:

```python
        ...
        context.js_config.enable_content_item_selection_mode.assert_called_once_with(
            form_action="http://example.com/module_item_configurations",
            form_fields=dict(form_params, authorization=bearer_token_schema.authorization_param.return_value)
        )

    @pytest.fixture
    def form_params(self):
        return {
            "user_id": "TEST_USER_ID",
            "resource_link_id": "TEST_RESOURCE_LINK_ID",
            "oauth_consumer_key": "TEST_OAUTH_CONSUMER_KEY",
            "tool_consumer_instance_guid": "TEST_TOOL_CONSUMER_INSTANCE_GUID",
            "context_id": "TEST_CONTEXT_ID",
        }

    @pytest.fixture
    def pyramid_request(self, pyramid_request, form_params):
        pyramid_request.params = dict(
            form_params,
            oauth_nonce="TEST_OAUTH_NONCE",
            oauth_timestamp="TEST_OAUTH_TIMESTAMP",
            oauth_signature="TEST_OAUTH_SIGNATURE",
        )
        
        return pyramid_request
```

Reply:
Gone ahead and done this: https://github.com/hypothesis/lms/pull/1546/commits/8162e84682102c01a6da3a742680eede8afb7cc1

It's cool that you can make a dict like this, btw: `dict(other_dict, **additional_items)`